lib/rspec/core/example.rb

```diff
@@ -248,6 +248,8 @@ def run(example_group_instance, reporter)
         @reporter = reporter
         RSpec.configuration.configure_example(self, hooks)
         RSpec.current_example = self
+        prev_scope = RSpec.current_scope
+        RSpec.current_scope = :example
```

Comment:
It seems to me this should go inside `with_around_and_singleton_context_hooks` block.

Reply:
It ended up inside `with_around_example_hooks` which is called inside `with_around_and_singleton_context_hooks` thanks to your suggestion.